spring-pulsar/src/main/java/io/streamnative/example/datagen/SignupGenerator.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.streamnative.example.datagen;
+
+import com.devskiller.jfairy.Fairy;
+import com.devskiller.jfairy.producer.person.Person;
+import io.streamnative.example.model.Signup;
+import io.streamnative.example.model.SignupTier;
+import java.util.Random;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SignupGenerator {
+
+  private final Fairy fairy;
+
+  public SignupGenerator() {
+    this.fairy = Fairy.create();
+  }
+
+  public Signup generate() {
+    Person person = fairy.person();
+    return new Signup(
+        generateSignupTier(),
+        person.getFirstName(),
+        person.getLastName(),
+        person.getTelephoneNumber(),
+        person.getCompany().getName(),
+        person.getCompanyEmail(),
+        System.currentTimeMillis());
+  }
+
+  private static SignupTier generateSignupTier() {
+    return SignupTier.values()[new Random().nextInt(SignupTier.values().length)];
```

Comment:
Should we store the random as static instance var to get better randoms?

Reply:
I was thinking about this too but also liked the simplicity of the one-liner as it's not really critical. I'll change it anyway for best practice reasons :)